public/app/features/alerting/unified/rule-list/filter/SavedSearches.tsx

```diff
@@ -0,0 +1,556 @@
+
+    case 'START_RENAME':
+      // Only start rename if no action is active
+      return state.activeAction === 'idle' ? { ...state, activeAction: { type: 'renaming', id: action.id } } : state;
+
+    case 'START_DELETE':
+      // Only start delete if no action is active
+      return state.activeAction === 'idle' ? { ...state, activeAction: { type: 'deleting', id: action.id } } : state;
+
+    case 'CANCEL_ACTION':
+    case 'COMPLETE_ACTION':
+      // Return to idle state
+      return { ...state, activeAction: 'idle' };
+
+    case 'APPLY_AND_CLOSE':
+      // Only apply if no action is active, then close
+      return state.activeAction === 'idle' ? { isOpen: false, activeAction: 'idle' } : state;
+
+    default:
+      return state;
+  }
+}
+
+// ============================================================================
+// Main Component
+// ============================================================================
+
+export function SavedSearches({
+  savedSearches,
+  currentSearchQuery,
+  onSave,
+  onRename,
+  onDelete,
+  onApply,
+  onSetDefault,
+  isLoading = false,
+  className,
+}: SavedSearchesProps) {
+  const styles = useStyles2(getStyles);
+
+  // Centralized state management via reducer
+  const [state, dispatch] = useReducer(dropdownReducer, initialState);
+  const { isOpen, activeAction } = state;
+
+  // Refs
+  const saveButtonRef = useRef<HTMLButtonElement>(null);
+  const dialogRef = useRef<HTMLDivElement>(null);
+
+  // Focus dialog when dropdown opens to enable keyboard navigation
+  useEffect(() => {
+    if (isOpen && activeAction === 'idle') {
+      // Small delay to ensure the dropdown is rendered
+      const timer = setTimeout(() => {
+        // Focus the dialog to capture keyboard events (like Escape)
+        // We focus the dialog instead of the save button because the button may be disabled
+        dialogRef.current?.focus();
+      }, 50);
+      return () => clearTimeout(timer);
+    }
+    return undefined;
+  }, [isOpen, activeAction]);
+
+  // Handle click outside to close dropdown - excludes portal elements (like action menu)
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (isOpen && dialogRef.current && event.target instanceof Node && !dialogRef.current.contains(event.target)) {
+        // Check if click is on a portal element (action menu dropdown)
+        if (event.target instanceof Element) {
+          const isPortalClick =
+            event.target.closest('[data-popper-placement]') || event.target.closest('[role="menu"]');
+
+          if (!isPortalClick) {
+            dispatch({ type: 'CLOSE' });
+          }
+        } else {
+          dispatch({ type: 'CLOSE' });
+        }
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
```

Comment:
@dhalachliyski this should use useClickAway (or better useFloating). 

Or use Dropdown component for the actual dropdown. 

the code mentions popper 

```
const isPortalClick =
            event.target.closest('[data-popper-placement]') || event.target.closest('[role="menu"]');

```

We do not use popper anymore, but useFloating